### ISSUE TITLE
Recorder issue with request exception

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbem#587]
I believe this and #588 are duplicates.

In testing with run_cimoperations.py, I generated an openenumerateInstances exception by using a host name that was invalid.  This generated an exception (ConnectionError). However, we were using the recorder and the call to the recorder generated a second exception.
```
Traceback (most recent call last):
  File "/home/kschopmeyer/pywbem/githubpywbem/pywbem/pywbem/cim_operations.py", line 3915, in OpenEnumerateInstances
    **extra)
  File "/home/kschopmeyer/pywbem/githubpywbem/pywbem/pywbem/cim_operations.py", line 866, in _imethodcall
    recorder=self.operation_recorder)
  File "/home/kschopmeyer/pywbem/githubpywbem/pywbem/pywbem/cim_http.py", line 684, in wbem_request
    raise ConnectionError('Error with file URL %s: %s' % (url, exc))
pywbem.exceptions.ConnectionError: Error with file URL localhost.py: [Errno 2] No such file or directory: 'localhost.  Note that we had not included the http:// so that the request was invalid.
```

It appears that the statement in _recorder.py at 410 & 411
```
            tc_http_request['url'] = TestClientRecorder.TESTCASE_URL + \
                http_request.target
```
cannot handle a NoneType in http_request.target. at this point.

During handling of the above exception, another exception occurred:

```
Traceback (most recent call last):
  File "testsuite/run_cim_operations.py", line 3191, in test_simple_iter_enum
    for inst in iterator:
  File "/home/kschopmeyer/pywbem/githubpywbem/pywbem/pywbem/cim_operations.py", line 1859, in IterEnumerateInstances
    MaxObjectCount=MaxObjectCount, **extra)
  File "/home/kschopmeyer/pywbem/githubpywbem/pywbem/pywbem/cim_operations.py", line 3923, in OpenEnumerateInstances
    self.operation_recorder.record_staged()
  File "/home/kschopmeyer/pywbem/githubpywbem/pywbem/pywbem/_recorder.py", line 300, in record_staged
    self.record(pwargs, pwresult, httpreq, httpresp)
  File "/home/kschopmeyer/pywbem/githubpywbem/pywbem/pywbem/_recorder.py", line 411, in record
    http_request.target
TypeError: Can't convert 'NoneType' object to str implicitly
```

**Comment by andy-maier** [comment]
This issue and issue #588 are not duplicates.
This one is caused by carelessly assuming the target URL path is set in the HttpRequest, which does not seem to be true in case of a connection error.
Will add a check for that.

**Comment by andy-maier** [comment]
Fixed with PR #612
